server/src/main/java/io/crate/execution/ddl/tables/TransportDropConstraintAction.java

```diff
@@ -0,0 +1,160 @@
+            DropConstraintRequest::new,
+            AcknowledgedResponse::new,
+            AcknowledgedResponse::new,
+            "drop-constraint");
+        this.nodeContext = nodeContext;
+        this.indicesService = indicesService;
+    }
+
+
+    @Override
+    public ClusterStateTaskExecutor<DropConstraintRequest> clusterStateTaskExecutor(DropConstraintRequest request) {
+        return new DDLClusterStateTaskExecutor<>() {
+            @Override
+            protected ClusterState execute(ClusterState currentState, DropConstraintRequest request) throws Exception {
+                Metadata.Builder metadataBuilder = Metadata.builder(currentState.metadata());
+                boolean updated = false;
+                if (request.isPartitioned()) {
+                    String templateName = PartitionName.templateName(request.relationName().schema(), request.relationName().name());
+                    IndexTemplateMetadata indexTemplateMetadata = currentState.metadata().templates().get(templateName);
+
+                    // Removal happens by key, value in the deepest map is not used.
+                    Map<String, Object> mapToRemove = Map.of("_meta", Map.of("check_constraints", Map.of(request.constraintName(), "")));
+
+                    IndexTemplateMetadata newIndexTemplateMetadata = DDLClusterStateHelpers.updateTemplate(
+                        indexTemplateMetadata,
+                        Collections.emptyMap(),
+                        mapToRemove,
+                        Settings.EMPTY,
+                        IndexScopedSettings.DEFAULT_SCOPED_SETTINGS // Not used if new settings are empty
+                    );
+                    updated = true;
+                    metadataBuilder.put(newIndexTemplateMetadata);
+                }
+
+                updated |= updateMapping(currentState, metadataBuilder, request);
+                if (updated) {
+                    // ensure the new table can still be parsed into a DocTableInfo to avoid breaking the table.
+                    new DocTableInfoFactory(nodeContext).create(request.relationName(), currentState);
+                    return ClusterState.builder(currentState).metadata(metadataBuilder).build();
+                }
+                return currentState;
+            }
+        };
+    }
+
+    private boolean updateMapping(ClusterState currentState,
+                                  Metadata.Builder metadataBuilder,
+                                  DropConstraintRequest request) throws IOException {
+        Index[] concreteIndices = resolveIndices(currentState, request.relationName().indexNameOrAlias());
+        boolean updated = false;
+
+        for (Index index : concreteIndices) {
+            final IndexMetadata indexMetadata = currentState.metadata().getIndexSafe(index);
+
+            Map<String, Object> currentSource = indexMetadata.mapping().sourceAsMap();
+
+            updated |= constraintRemoved(currentSource, request.constraintName());
```

Comment:
was not sure whether updated = true, when ALL or at least one. For non-partitioned tables it's always one index(?) so it doesn't matter here. 

In general, I added this updated flag to avoid unnecessary 
`ClusterState.builder(currentState).metadata(metadataBuilder).build()` calls but I think we already cut non-existing check scenario in analyzer, so can just remove this flag thingy both from here and above. wdyt?

Reply:
> was not sure whether updated = true, when ALL or at least one. For non-partitioned tables it's always one index(?) so it doesn't matter here.

If at least one changes the state, updated must be true (like your code does). Or do I misunderstand your comment?

Due to concurrent calls, the analyzer state could be different to the actual cluster state. So this code could be called even that the constraint is already remove. So avoiding unnecessary state changed is good to do.

Reply:
> at least one

yep, was thinking the same. the only reason about thinking about ALL was that having several indices and not finding constraint in one of them it's not an expected, isn't it? maybe need at least `WARN` about it - but not sure this case is even possible.